src/gui/statusbar.cpp

```diff
@@ -106,63 +107,58 @@ StatusBar::StatusBar(QStatusBar *bar)
     m_altSpeedsBtn->setMaximumWidth(28 + 6);
 
-    m_statusSep1 = new QFrame(bar);
-    m_statusSep1->setFrameStyle(QFrame::VLine);
-    m_statusSep1->setFrameShadow(QFrame::Raised);
-    m_statusSep2 = new QFrame(bar);
-    m_statusSep2->setFrameStyle(QFrame::VLine);
-    m_statusSep2->setFrameShadow(QFrame::Raised);
-    m_statusSep3 = new QFrame(bar);
-    m_statusSep3->setFrameStyle(QFrame::VLine);
-    m_statusSep3->setFrameShadow(QFrame::Raised);
-    m_statusSep4 = new QFrame(bar);
-    m_statusSep4->setFrameStyle(QFrame::VLine);
-    m_statusSep4->setFrameShadow(QFrame::Raised);
-    m_layout->addWidget(m_DHTLbl);
-    m_layout->addWidget(m_statusSep1);
-    m_layout->addWidget(m_connecStatusLblIcon);
-    m_layout->addWidget(m_statusSep2);
-    m_layout->addWidget(m_altSpeedsBtn);
-    m_layout->addWidget(m_statusSep4);
-    m_layout->addWidget(m_dlSpeedLbl);
-    m_layout->addWidget(m_statusSep3);
-    m_layout->addWidget(m_upSpeedLbl);
+    QFrame *statusSep1 = new QFrame(this);
+    statusSep1->setFrameStyle(QFrame::VLine);
+    statusSep1->setFrameShadow(QFrame::Raised);
+    QFrame *statusSep2 = new QFrame(this);
+    statusSep2->setFrameStyle(QFrame::VLine);
+    statusSep2->setFrameShadow(QFrame::Raised);
+    QFrame *statusSep3 = new QFrame(this);
+    statusSep3->setFrameStyle(QFrame::VLine);
+    statusSep3->setFrameShadow(QFrame::Raised);
+    QFrame *statusSep4 = new QFrame(this);
+    statusSep4->setFrameStyle(QFrame::VLine);
+    statusSep4->setFrameShadow(QFrame::Raised);
+    layout->addWidget(m_DHTLbl);
+    layout->addWidget(statusSep1);
+    layout->addWidget(m_connecStatusLblIcon);
+    layout->addWidget(statusSep2);
+    layout->addWidget(m_altSpeedsBtn);
+    layout->addWidget(statusSep4);
+    layout->addWidget(m_dlSpeedLbl);
+    layout->addWidget(statusSep3);
+    layout->addWidget(m_upSpeedLbl);
 
-    bar->addPermanentWidget(m_container);
-    bar->setStyleSheet("QWidget {margin: 0;}");
-    m_container->adjustSize();
-    bar->adjustSize();
+    addPermanentWidget(container);
+    setStyleSheet("QWidget {margin: 0;}");
+    container->adjustSize();
+    adjustSize();
     // Is DHT enabled
     m_DHTLbl->setVisible(session->isDHTEnabled());
-    refreshStatusBar();
-    connect(BitTorrent::Session::instance(), &BitTorrent::Session::statsUpdated
-            , this, &StatusBar::refreshStatusBar);
+    refresh();
+    connect(session, &BitTorrent::Session::statsUpdated, this, &StatusBar::refresh);
 }
 
 StatusBar::~StatusBar()
 {
     qDebug() << Q_FUNC_INFO;
 }
 
-QPushButton* StatusBar::connectionStatusButton() const
-{
-    return m_connecStatusLblIcon;
-}
-
 void StatusBar::showRestartRequired()
 {
     // Restart required notification
     const QString restartText = tr("qBittorrent needs to be restarted");
-    QLabel *restartIconLbl = new QLabel(m_bar);
+    QLabel *restartIconLbl = new QLabel(this);
     restartIconLbl->setPixmap(QPixmap(":/icons/qbt-theme/dialog-warning.png").scaled(QSize(24,24)));
```

Comment:
`this->style()->standardIcon(QStyle::SP_MessageBoxWarning)` might be used (yes, I don't understand why do we need those two "dialog-" icons).